lib/Dialect/LWE/Transforms/AddDebugPort.cpp

```diff
@@ -73,22 +73,33 @@ LogicalResult insertExternalCall(func::FuncOp op, Type lwePrivateKeyType) {
   // implicit assumption the first argument is private key
   auto privateKey = op.getArgument(0);
 
-  ImplicitLocOpBuilder b =
-      ImplicitLocOpBuilder::atBlockBegin(module.getLoc(), module.getBody());
+  ImplicitLocOpBuilder b = ImplicitLocOpBuilder::atBlockBegin(
+      op.getLoc(), &op.getBody().getBlocks().front());
+
+  auto insertCall = [&](Value value) {
+    Type valueType = value.getType();
+    if (auto lweCiphertextType = dyn_cast<NewLWECiphertextType>(valueType)) {
```

Comment:
this won't work for arguments that have some shaped type of a ciphertext, right? maybe add a quick comment here, but i assume we will eventually have tensors of ciphertexts after we figure out layouts properly

Reply:
Added the comment.